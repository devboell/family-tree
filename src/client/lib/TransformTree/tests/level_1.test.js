import {
  onePartnerOneChild,
  onePartnerTwoChildren,
  family1,
  family2,
  family3,
} from '../test_data/level_1'
import { transformTree } from '../index'

const dimensions = {
  boxWidth: 80,
  boxHeight: 80,
  marginX: 20,
  marginY: 20,
}

it('should transform onePartnerOneChild', () => {
  expect(transformTree(onePartnerOneChild, dimensions)).toMatchSnapshot()
})

it('should transform onePartnerOneChild', () => {
  expect(transformTree(onePartnerTwoChildren, dimensions)).toMatchSnapshot()
})

it('should transform family1', () => {
  expect(transformTree(family1, dimensions)).toMatchSnapshot()
})

it('should transform family2', () => {
  expect(transformTree(family2, dimensions)).toMatchSnapshot()
})

it('should transform family3', () => {
  expect(transformTree(family3, dimensions)).toMatchSnapshot()
})
