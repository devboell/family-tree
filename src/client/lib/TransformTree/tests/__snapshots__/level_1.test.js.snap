// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should transform family1 1`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 2 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 200,
      "value": "root - partner 1",
    },
    Object {
      "children": Array [],
      "level": 0,
      "totalHeight": 100,
      "totalWidth": 200,
      "value": "root - partner 2",
    },
  ],
  "totalHeight": 200,
  "totalWidth": 400,
  "value": "root",
}
`;

exports[`should transform family2 1`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 2 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 200,
      "value": "root - partner 1",
    },
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 2 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 200,
      "value": "root - partner 2",
    },
  ],
  "totalHeight": 200,
  "totalWidth": 400,
  "value": "root",
}
`;

exports[`should transform family3 1`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 2 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 3 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 300,
      "value": "root - partner 1",
    },
    Object {
      "children": Array [],
      "level": 0,
      "totalHeight": 100,
      "totalWidth": 200,
      "value": "partner 2",
    },
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 2 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 200,
      "value": "root - partner 3",
    },
  ],
  "totalHeight": 200,
  "totalWidth": 700,
  "value": "root",
}
`;

exports[`should transform onePartnerOneChild 1`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 200,
      "value": "root - partner",
    },
  ],
  "totalHeight": 200,
  "totalWidth": 200,
  "value": "root",
}
`;

exports[`should transform onePartnerOneChild 2`] = `
Object {
  "level": 0,
  "partners": Array [
    Object {
      "children": Array [
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 1 - level 1",
        },
        Object {
          "level": 1,
          "partners": Array [],
          "totalHeight": 100,
          "totalWidth": 100,
          "value": "child 2 - level 1",
        },
      ],
      "level": 0,
      "totalHeight": 200,
      "totalWidth": 200,
      "value": "root - partner",
    },
  ],
  "totalHeight": 200,
  "totalWidth": 200,
  "value": "root",
}
`;
